src/main/java/com/aayushatharva/sourcecenginequerycacher/gameserver/a2splayer/PlayerClient.java

```diff
@@ -31,15 +31,14 @@ public void run() {
                     .option(ChannelOption.ALLOCATOR, PooledByteBufAllocator.DEFAULT)
                     .option(ChannelOption.SO_SNDBUF, Config.SendBufferSize)
                     .option(ChannelOption.SO_RCVBUF, Config.ReceiveBufferSize)
-                    .option(EpollChannelOption.MAX_DATAGRAM_PAYLOAD_SIZE, 1400)

```

Comment:
We need some sort of limiter in packet size here. 
What happens with a full 100player server is that packet size grows beyond the 1400bytes that valve proposed in the protocol. 

and if you’re on the same machine the cacher can actually receive a say 1580byte udp datagram internally. 
But as soon as you try and send that over the internet it gets fragmented. And fragmented UDP packets get dropped at router level in 99% of today’s routers. 


so we either implement a proper multi packet handling as it is proposed in the protocol, or we truncate the packet down to 1400 bytes

And just to make sure We should do that for all three packet types.

Reply:
Alright then. Let's restore it.